File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    answer = []
    for i in range(len(arr)):
        if flag[i]:
            for a in range(arr[i]*2):
                answer.append(arr[i])
        else:
            for b in range(arr[i]):
                del answer[-1]
                
    return answer

File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
    let result = [];
    for(let i=0;i<flag.length;i++){
        if(flag[i]){
            for(let x=0;x<arr[i]*2;x++){
                result.push(arr[i]);
            }
        }else{
            for(let x=0;x<arr[i];x++){
                result.pop();
            }
        }
    }
    return result;
}